src/ConfigCatClient.php

```diff
@@ -179,31 +194,44 @@ public function getValueDetails(string $key, mixed $defaultValue, ?User $user =
         try {
             $settingsResult = $this->getSettings();
             if (empty($settingsResult->settings)) {
-                $message = "Config JSON is not present. Returning defaultValue: '" . $defaultValue . "'.";
-                $this->logger->error($message);
-                $details = EvaluationDetails::fromError($key, $defaultValue, $user, $message);
+                $message = "Config JSON is not present. Returning the `{DEFAULT_PARAM_NAME}` parameter that you specified in your application: '{DEFAULT_PARAM_VALUE}'.";
+                $messageCtx = [
+                    'event_id' => 1000,
+                    'DEFAULT_PARAM_NAME' => '$defaultValue', 'DEFAULT_PARAM_VALUE' => Utils::getStringRepresentation($defaultValue)
+                ];
+                $this->logger->error($message, $messageCtx);
+                $details = EvaluationDetails::fromError($key, $defaultValue, $user, InternalLogger::format($message, $messageCtx));
                 $this->hooks->fireOnFlagEvaluated($details);
                 return $details;
             }
 
             if (!array_key_exists($key, $settingsResult->settings)) {
-                $message = "Evaluating getValueDetails('" . $key . "') failed. " .
-                    "Value not found for key " . $key . ". " .
-                    "Returning defaultValue: " . Utils::getStringRepresentation($defaultValue) . ". " .
-                    "Here are the available keys: " . implode(", ", array_keys($settingsResult->settings));
-                $this->logger->error($message);
-                $details = EvaluationDetails::fromError($key, $defaultValue, $user, $message);
+                $message = "Failed to evaluate setting '{KEY}' (the key was not found in config JSON). " .
+                    "Returning the `{DEFAULT_PARAM_NAME}` parameter that you specified in your application: '{DEFAULT_PARAM_VALUE}'. " .
+                    "Available keys: {AVAILABLE_KEYS}.";
+                $messageCtx = [
+                    'event_id' => 1001,
+                    'KEY' => $key,
+                    'DEFAULT_PARAM_NAME' => '$defaultValue', 'DEFAULT_PARAM_VALUE' => Utils::getStringRepresentation($defaultValue),
+                    'AVAILABLE_KEYS' => implode(", ", array_keys($settingsResult->settings))
```

Comment:
Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`